src/commonMain/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumClient.kt

```diff
@@ -52,11 +53,11 @@ internal data class BroadcastStatus(val connection: Connection) : ElectrumClient
  *
  */
 internal sealed class ClientState {
```

Comment:
Why not extend `LoggingContext` here as in `ChanelState` ?

Reply:
Two reasons:
- The `Logger` should in my opinion not be part of the state data of  an FSM. That is the point I was making here: https://github.com/ACINQ/lightning-kmp/pull/355#discussion_r973153903. If anything we should make `ChannelState` be like `ClientState`, not the opposite. But in the case of the `ChannelState`, the top-level structure already contained a reference to `nodeParams`, so it was really less effort to do it there. Contrast to `ClientState` which is just an interface, we would have to explicitly add the logger to it.
- In the case of the channel, the "machine" is actually the `Peer`, so it would be quite messy to have the `Peer` pass a channel-related `Logger` to `ChannelState`. It's different from the couple `ElectrumState`/`ClientState` where states do not appear outside of the FSM.